rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -152,6 +157,135 @@ void voteFeePerKbChange_success() {
+
+        assertEquals(WhitelistResponseCode.SUCCESS.getCode(), result);
+    }
+
+    @Test
+    void removeLockWhitelistAddress() {
+        WhitelistSupport whitelistSupport = mock(WhitelistSupportImpl.class);
+        Transaction tx = mock(Transaction.class);
+        String address = "address";
+        when(whitelistSupport.removeLockWhitelistAddress(tx, address)).thenReturn(WhitelistResponseCode.SUCCESS.getCode());
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+            .withWhitelistSupport(whitelistSupport)
+            .build();
+
+        int result = bridgeSupport.removeLockWhitelistAddress(tx, address);
+
+        assertEquals(WhitelistResponseCode.SUCCESS.getCode(), result);
+    }
+
+    @Test
+    void setLockWhitelistDisableBlockDelay() throws BlockStoreException, IOException {
+        // Set of Variables to be use in setLockWhitelistDisableBlockDelay
+        WhitelistSupport whitelistSupport = mock(WhitelistSupportImpl.class);
+        Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, WhitelistCaller.AUTHORIZED.getRskAddress());
+        BigInteger disableBlockDelayBI = BigInteger.ONE;
+        when(whitelistSupport.setLockWhitelistDisableBlockDelay(tx, disableBlockDelayBI, 0)).thenReturn(WhitelistResponseCode.SUCCESS.getCode());
+
+        // Set of variables to be use in bridgeSupportBuilder
+        BridgeStorageProvider provider = mock(BridgeStorageProvider.class);
+        Repository rskRepository = mock(Repository.class);
+        BridgeConstants bridgeConstantsMainNet = BridgeMainNetConstants.getInstance();
+        BtcBlockStoreWithCache.Factory btcBlockStoreFactory = mock(BtcBlockStoreWithCache.Factory.class);
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+            .withWhitelistSupport(whitelistSupport)
+            .withProvider(provider)
+            .withRepository(rskRepository)
+            .withBridgeConstants(bridgeConstantsMainNet)
+            .withBtcBlockStoreFactory(btcBlockStoreFactory)
+            .withActivations(activations)
+            .build();
+
+        // Set of variables to be used mocking
+        BtcBlockStoreWithCache btcBlockStore = mock(BtcBlockStoreWithCache.class);
+        StoredBlock storedBlock = mock(StoredBlock.class);
+        BtcBlock btcBlock = mock(BtcBlock.class);
+        NetworkParameters btcParams = bridgeConstantsMainNet.getBtcParams();
+        BtcBlock genesisBtcBlock = btcParams.getGenesisBlock();
+        Sha256Hash genesisBtcBlockHash = genesisBtcBlock.getHash();
+
+        when(btcBlockStore.getChainHead()).thenReturn(storedBlock);
+        when(storedBlock.getHeader()).thenReturn(btcBlock);
+        when(btcBlock.getHash()).thenReturn(genesisBtcBlockHash);
+        when(btcBlockStoreFactory.newInstance(rskRepository, bridgeConstantsMainNet, provider, activations)).thenReturn(btcBlockStore);
```

Comment:
```suggestion
Transaction tx = mock(Transaction.class);
        BigInteger disableBlockDelayBI = BigInteger.ONE;
        when(whitelistSupport.setLockWhitelistDisableBlockDelay(tx, disableBlockDelayBI, 0)).thenReturn(WhitelistResponseCode.SUCCESS.getCode());
        
        BridgeSupport bridgeSupport = bridgeSupportBuilder
            .withWhitelistSupport(whitelistSupport)
            .build();
```
Can't this be simplificated like this?

Reply:
Hey, @marcos-iov!
The rest of the code is needed to avoid a chain of Null Pointer Exceptions. If we remove the code suggested, we can get different issues that were fixed.

Reply:
pretty sure marcos is right, something like this:

```
        Transaction tx = mock(Transaction.class);
        int btcBlockchainBestChainHeight = 0;
        BigInteger disableBlockDelayBI = BigInteger.ONE;
        
        WhitelistSupport whitelistSupport = mock(WhitelistSupportImpl.class);
        when(whitelistSupport.setLockWhitelistDisableBlockDelay(tx, disableBlockDelayBI, btcBlockchainBestChainHeight)).thenReturn(WhitelistResponseCode.SUCCESS.getCode());

        BridgeSupport bridgeSupport = bridgeSupportBuilder
            .withWhitelistSupport(whitelistSupport)
            .build();
        when(bridgeSupport.getBtcBlockchainBestChainHeight()).thenReturn(btcBlockchainBestChainHeight);

        int result = bridgeSupport.setLockWhitelistDisableBlockDelay(tx, disableBlockDelayBI);
        assertEquals(WhitelistResponseCode.SUCCESS.getCode(), result);
```

should work

Reply:
When I put this piece of code, it throws an error:

```
Cannot invoke "co.rsk.peg.BtcBlockStoreWithCache.getChainHead()" because "this.btcBlockStore" is null
java.lang.NullPointerException: Cannot invoke "co.rsk.peg.BtcBlockStoreWithCache.getChainHead()" because "this.btcBlockStore" is null
```

Once I solve this issue, I get a second NPE, and so on. Part of my solution is to solve these errors.

Reply:
mmm, this part
`when(bridgeSupport.getBtcBlockchainBestChainHeight()).thenReturn(btcBlockchainBestChainHeight);`
should fix it.

Reply:
It doesn't fix the issue. 
<img width="1418" alt="image" src="https://github.com/rsksmart/rskj/assets/160710714/c4159e19-04c2-4997-bef0-fe92ff1bdaf9">

Reply:
you are right wilmer, the problem is the `getBtcBlockchainBestChainHeight()` method